### PR TITLE
Issue #16: Drop PHP 5.6 support

### LINKED ISSUE #16: Drop PHP 5.6 support

### PR DESCRIPTION
**Pull request by tim-bezhashvyly**
Closes #16.

### CODE REVIEW

**Comment by tim-bezhashvyly** [comment]
I had to re-configure build settings at Scrutinizer. Tests are now passing https://scrutinizer-ci.com/g/lizards-and-pumpkins/lib-image-processing-gd/inspections/70cf6c0a-c9b4-4418-b318-ce6ba177fc1b

File: tests/Unit/Suites/GdInscribeStrategyTest.php [hunk 1]
```diff
@@ -22,15 +23,14 @@ protected function setUp()
     
     public function testImageProcessorStrategyInterfaceIsImplemented()
     {
-        $strategy = new GdInscribeStrategy(1, 1, 'none');
+        $strategy = new GdInscribeStrategy(1, 1, 0);
         $this->assertInstanceOf(ImageProcessingStrategy::class, $strategy);
```
Comment (Schrank) [inline tests/Unit/Suites/GdInscribeStrategyTest.php]: you could do

    $mock = $this->getMock(GdInscribeStrategy::class);
    $this->assertInstanceOf(ImageProcessingStrategy::class, $mock);

then you don't have to care about the parameters

File: tests/Unit/Suites/GdInscribeStrategyTest.php [hunk 2]
```diff
@@ -22,15 +23,14 @@ protected function setUp()
...
     }
 
     public function testExceptionIsThrownIfWidthIsNotAnInteger()
     {
-        $this->expectException(InvalidImageDimensionException::class);
-        $this->expectExceptionMessage('Expected integer as image width, got string.');
-        (new GdInscribeStrategy('foo', 1, 0))->processBinaryImageData('');
+        $this->expectException(\TypeError::class);
+        new GdInscribeStrategy('foo', 1, 0);
```
Comment (Schrank) [inline tests/Unit/Suites/GdInscribeStrategyTest.php]: PHP is doing this, let's just remove the test.

File: tests/Unit/Suites/GdInscribeStrategyTest.php [hunk 3]
```diff
@@ -42,9 +42,8 @@ public function testExceptionIsThrownIfWidthIsNotPositive()
 
     public function testExceptionIsThrownIfHeightIsNotAnInteger()
     {
-        $this->expectException(InvalidImageDimensionException::class);
-        $this->expectExceptionMessage('Expected integer as image height, got string.');
-        (new GdInscribeStrategy(1, 'foo', 0))->processBinaryImageData('');
+        $this->expectException(\TypeError::class);
+        new GdInscribeStrategy(1, 'foo', 0);
```
Comment (Schrank) [inline tests/Unit/Suites/GdInscribeStrategyTest.php]: PHP is doing this, let's just remove the test.

File: tests/Unit/Suites/GdInscribeStrategyTest.php [hunk 4]
```diff
@@ -54,9 +53,9 @@ public function testExceptionIsThrownIfHeightIsNotPositive()
         (new GdInscribeStrategy(1, -1, 0))->processBinaryImageData('');
     }
 
-    public function testExceptionIsThrownIfInvalidBackgroundColorIsSpecified()
+    public function testExceptionIsThrownIfBackgroundColorIsNotAnInteger()
     {
-        $this->expectException(InvalidColorException::class);
+        $this->expectException(\TypeError::class);
         (new GdInscribeStrategy(1, 1, 'red'))->processBinaryImageData('');
```
Comment (Schrank) [inline tests/Unit/Suites/GdInscribeStrategyTest.php]: PHP is doing this, let's just remove the test.

File: tests/Unit/Suites/GdResizeStrategyTest.php [hunk 5]
```diff
@@ -27,9 +29,8 @@ public function testImageProcessorStrategyInterfaceIsImplemented()
 
     public function testExceptionIsThrownIfWidthIsNotAnInteger()
     {
-        $this->expectException(InvalidImageDimensionException::class);
-        $this->expectExceptionMessage('Expected integer as image width, got string.');
-        (new GdResizeStrategy('foo', 1))->processBinaryImageData('');
+        $this->expectException(\TypeError::class);
+        new GdResizeStrategy('foo', 1);
```
Comment (Schrank) [inline tests/Unit/Suites/GdResizeStrategyTest.php]: PHP is doing this, let's just remove the test.

File: tests/Unit/Suites/GdResizeStrategyTest.php [hunk 6]
```diff
@@ -41,9 +42,8 @@ public function testExceptionIsThrownIfWidthIsNotPositive()
 
     public function testExceptionIsThrownIfHeightIsNotAnInteger()
     {
-        $this->expectException(InvalidImageDimensionException::class);
-        $this->expectExceptionMessage('Expected integer as image height, got string.');
-        (new GdResizeStrategy(1, 'foo'))->processBinaryImageData('');
+        $this->expectException(\TypeError::class);
+        new GdResizeStrategy(1, 'foo');
```
Comment (Schrank) [inline tests/Unit/Suites/GdResizeStrategyTest.php]: PHP is doing this, let's just remove the test.

File: tests/Unit/Suites/GdInscribeStrategyTest.php (hunk 1 above)
Comment (tim-bezhashvyly) [inline tests/Unit/Suites/GdInscribeStrategyTest.php]: I don't think this will work.

And also `getMock` is deprecated. 😉

**Pull request merged**